mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1074,51 +976,106 @@ where
     Box::new(fut.boxed().compat())
 }
 
-pub fn search_for_swap_tx_spend_my<T>(
-    coin: &T,
+pub fn check_if_my_payment_completed<T>(
+    coin: T,
     time_lock: u32,
     other_pub: &[u8],
     secret_hash: &[u8],
-    tx: &[u8],
-    search_from_block: u64,
-) -> Result<Option<FoundSwapTxSpend>, String>
+) -> Box<dyn Future<Item = (), Error = String> + Send>
 where
-    T: AsRef<UtxoArc> + UtxoCoinCommonOps,
+    T: AsRef<UtxoCoinFields> + UtxoCommonOps + Send + Sync + 'static,
 {
-    coin.search_for_swap_tx_spend(
+    let fut = check_if_my_payment_sent(coin, time_lock, other_pub, secret_hash).and_then(|tx| {
```

Comment:
As we first wait for payment confirmation during swap *before* check_if_my_payment_completed, we can simply return `Ok` here as it is possible but not very likely (with decent number of confirmations) that confirmed transaction won't be present on blockchain.

Also 1 more argument against this: in native utxo mode `check_if_my_payment_sent` sends heavy `listreceivedbyaddress` call which can cause daemon RPC to stuck if many concurrent swaps are running. Nnative daemon RPC can process only 1 request in parallel and single `listreceivedbyaddress` takes 300+ ms to be processed. Imagine if there're hundreds of such requests.

Reply:
Ok, sounds right. I've added this additional checking just to be sure that the transaction is on blockchain.

Reply:
Done